### ISSUE TITLE
NOTE: Redux Form's author advises to use React Final Form instead

### ISSUE THREAD

**Issue by micschk** [silverstripe/silverstripe-admin#1266]
_(probably already on the radar but leaving here just in case it was not)_  

On trying to update some modules to work with react rendered forms I noticed the below info [on the Redux Form repo](https://github.com/redux-form/redux-form/#%EF%B8%8F-attention-%EF%B8%8F)

> ## ⚠️ ATTENTION ⚠️
> 
> If you're just getting started with your application and are looking for a form solution, the [general consensus of the community](https://twitter.com/acemarke/status/1124771065115185152) is that you should _not_ put your form state in Redux. The author of Redux Form took all of the lessons he learned about form use cases from maintaining Redux Form and built [🏁 React Final Form](https://github.com/final-form/react-final-form#-react-final-form), which he recommends you use if you are just starting your project. It's also pretty easy to migrate to from Redux Form, because the `<Field>` component APIs are so similar. [Here is a blog post](https://codeburst.io/final-form-the-road-to-the-checkered-flag-cd9b75c25fe) where he explains his reasoning, or [there are two talks](https://github.com/final-form/react-final-form#videos) if you prefer video. [Formik](https://jaredpalmer.com/formik/) is also a nice solution.
> 
> The only good reason, [in the author's view](https://twitter.com/erikras/status/1035082880341483520), to use Redux Form in your application is if you need _really_ tight coupling of your form data with Redux, specifically if you need to subscribe to it and modify it from parts of your application far from your form component, e.g. on another route. If you don't have that requirement, use [🏁 React Final Form](https://github.com/final-form/react-final-form#-react-final-form).

Just in case Silverstripe is using davidkpiano/react-redux-form instead(?), [that shows a similar notice](https://github.com/davidkpiano/react-redux-form#%EF%B8%8F-this-project-is-in-maintenance-mode-only-please-consider-using-formik-instead):

> ### ⚠️ This project is in maintenance mode only. Please consider using [Formik](https://github.com/jaredpalmer/formik) instead.

Adding this issue here, closing previously opened issue on [silverstripe/silverstripe-framework](https://github.com/silverstripe/silverstripe-framework/issues/10149).

**Comment by unclecheese** [comment]
Yeah, this package was abandoned shortly after SS4 was released. Gotta love frontend dev. 

The reasoning behind keeping form state out of your app state is pretty sound. [React hook form](https://react-hook-form.com/) is really the best thing out there now, but we're several major refactors away from being able to extract all that logic out of redux.

**Comment by GuySartorelli** [comment]
[`@tanstack/react-form`](tanstack.com/form) maybe a suitable alternative if we're willing to rework the way we do things a bit. Lots of commercial supporters and lots of maintainers, which suggests it'll have good longevity - and it's not specifically react which might make it easier to work with if/when we do implement https://github.com/silverstripe/.github/issues/346
